stock_request_purchase/tests/test_stock_request_purchase.py

```diff
@@ -8,15 +8,14 @@
 
 
 class TestStockRequestPurchase(TestStockRequest):
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        cls.route_buy = cls.warehouse.buy_pull_id.route_id
-        cls.supplier = cls.env["res.partner"].create({"name": "Supplier"})
-        cls.product.write(
+    def setUp(self):
```

Comment:
This change is not correct. Please revert it.

Reply:
Hello @pedrobaeza,
I've reverted the change as requested. The original setup used setUpClass(cls), which caused an AttributeError because cls.warehouse wasn't initialized properly. To address this, I changed it to setUp(self), ensuring the superclass's setUp method was called to initialize the attributes.
As I'm new to Odoo development, could you please advise me on the correct way to initialize these attributes in setUpClass without causing errors?
Thank you for your help.

Reply:
OK, I see. The problem is that the original test is also with `setUp`:

https://github.com/OCA/stock-logistics-request/blob/17.0/stock_request/tests/test_stock_request.py#L13

which is not correct at all, as it's not making use of this method that avoids to execute the setup for each of the test methods, running only once instead of N times.

Can you please make a new PR changing the test of `stock_request` to setUpClass?

Reply:
> Can you please make a new PR changing the test of `stock_request` to setUpClass?

Agree. I would say in addition of this, moving the setupClass to a common.py is a best practice too.

Reply:
Hi @pedrobaeza 
I noticed that the changes requested for the stock_request module tests in v17 are already done in v16:
https://github.com/OCA/stock-logistics-request/blob/16.0/stock_request/tests/test_stock_request.py
But they are missing in v17: 
https://github.com/OCA/stock-logistics-request/blob/17.0/stock_request/tests/test_stock_request.py

Could you please confirm if the changes you want me to make are the same as those made in v16? If so, do you know why these changes are not in v17?

Additionally, should I create a separate PR for the modifications in stock_request, or should I include these changes in another commit within this existing PR I have for migrating stock_request_purchase?

Thanks!

Reply:
Yes, that's it. You can do it as a separate commit, but inside this PR, as it's an adjustment linked to this migration.

Reply:
Hi @pedrobaeza , all done. 
Could you please review it?
Thanks